pkg/server/router/did.go

```diff
@@ -87,8 +87,15 @@ func (dr DIDRouter) CreateDIDByMethod(ctx context.Context, w http.ResponseWriter
 	}
 
 	var request CreateDIDByMethodRequest
+	invalidCreateDIDRequest := "invalid create DID request"
 	if err := framework.Decode(r, &request); err != nil {
```

Comment:
alternativly you could do 

if framework.Decode(r, &request) != nil || framework.ValidateRequest(request) != nil {
    errMsg := invalidCreateDIDRequest
		logrus.WithError(err).Error(errMsg)
		return framework.NewRequestError(errors.Wrap(err, errMsg), http.StatusBadRequest)
}

since its the same output (but you need to get err variable in a golang way somehow lol maybe not possible)

Reply:
yeah, unfortunately this wouldn't work because both return unique errors and we couldn't capture both. conditional error assignment would be a nice feature 🤔